src/features/donations/components/treeDonation/GiftForm.tsx

```diff
@@ -75,37 +81,64 @@ export default function GiftForm({
               </span>
             )}
           </div>
-        </div>
-        <div className={styles.formRow}>
+        </div> <br />
+        {showEmail ? (
+          <>
+          <div className={styles.giftLine}>
+            <div className={styles.singleGiftTitle}>
+            {t('donate:giftRecipient')}
+          </div>
+          <button
+            onClick={() => {setshowEmail(false); giftDetails.email = null} }
+            className={styles.singleGiftRemove}
+          >
+            {t('donate:removeRecipient')}
+          </button>
+            </div>
           <div style={{ width: '100%' }}>
-            <MaterialTextField
-              name={'email'}
-              onChange={changeGiftDetails}
-              label={t('donate:email')}
-              variant="outlined"
-              inputRef={register({
-                required: true,
-                pattern: /^([a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?)$/i,
-              })}
-            />
-            {errors.email && (
-              <span className={styles.formErrors}>
-                {t('donate:emailRequired')}
-              </span>
-            )}
+              <MaterialTextField
+                multiline
+                rowsMax="4"
+                label={t('donate:giftMessage')}
+                variant="outlined"
+                name={'giftMessage'}
```

Comment:
The input given for the gift message always gets lost when navigating back for this form if the gift email was entered and also if no gift email was specified:

<img width="408" alt="Bildschirmfoto 2021-06-15 um 09 42 28" src="https://user-images.githubusercontent.com/1532418/122013897-ebe3fe80-cdbe-11eb-8e5c-e91c474bb931.png">

Reply:
@harshvitra maybe if you have time you can work on this to get this pull request finished?